<launch>
  <arg name="rviz" default="true"/>
  <arg name="rviz_config" default="$(find-pkg-share nav_sim)/rviz/nav_sim.rviz"/>
  <arg name="config" default="$(find-pkg-share nav_sim)/config/landmark_config.yaml"/>
  <arg name="nav_sim_param" default="$(find-pkg-share nav_sim)/config/nav_sim.param.yaml"/>

  <node name="world_to_map" pkg="tf2_ros" exec="static_transform_publisher" args="0 0 0 0 0 0 world map"/>

  <node name="nav_sim" pkg="nav_sim" exec="nav_sim_node" output="screen">
    <param name="config" value="$(var config)"/>
    <param from="$(var nav_sim_param)"/>
    <remap from="current_pose" to="/localization/pose"/>
    <remap from="current_pose_with_covariance" to="/localization/pose_with_covariance"/>
  </node>

  <group if="$(var rviz)">
    <node name="rviz2" pkg="rviz2" exec="rviz2" args="-d $(var rviz_config)"/>
  </group>
</launch>
